signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java

```diff
@@ -54,25 +56,74 @@ public MappedResults<ArtifactSigner> load(
-    final Map<Path, String> configFilesContentMap = configFilesMapPair.getLeft();
-    final int configFileErrorCount = configFilesMapPair.getRight();
 
-    final String timeTaken =
-        DurationFormatUtils.formatDurationHMS(Duration.between(start, Instant.now()).toMillis());
     LOG.info(
         "Signer configuration metadata files read in memory {} in {}",
-        configFilesContentMap.size(),
-        timeTaken);
+        configFileContent.getContentMap().size(),
+        calculateTimeTaken(start));
+
+    final Instant conversionStartInstant = Instant.now();
+    // convert yaml file content to list of SigningMetadata
+    final MappedResults<SigningMetadata> signingMetadataResults =
+        convertConfigFileContent(configFileContent.getContentMap(), signerParser);
+
+    LOG.debug(
+        "Signing configuration metadata files converted to signing metadata {}",
+        signingMetadataResults.getValues().size());
+
+    final Collection<SigningMetadata> signingMetadata = signingMetadataResults.getValues();
+    // filter hashicorp signing metadata, we want them to be processed sequentially.
+    final Map<Boolean, List<SigningMetadata>> partitionedMetadata =
+        signingMetadata.stream()
+            .collect(
+                Collectors.partitioningBy(metadata -> !metadata.getType().equals("hashicorp")));
+    final MappedResults<ArtifactSigner> metadataResultParallel =
+        mapToArtifactSignerInParallel(partitionedMetadata.get(true), signerParser);
+    final MappedResults<ArtifactSigner> metadataResultSequential =
+        mapToArtifactSignerSequentially(partitionedMetadata.get(false), signerParser);
 
     final MappedResults<ArtifactSigner> metadataResult =
-        processMetadataFilesInParallel(configFilesContentMap, signerParser);
-    metadataResult.mergeErrorCount(configFileErrorCount);
+        MappedResults.merge(metadataResultParallel, metadataResultSequential);
+
+    // merge error counts of config file parsing errors ...
+    metadataResult.mergeErrorCount(signingMetadataResults.getErrorCount());
+    metadataResult.mergeErrorCount(configFileContent.getErrorCount());
+
+    LOG.info(
+        "Total Artifact Signer loaded via configuration files: {}\nError count {}\nTime Taken: {}.",
+        metadataResult.getValues().size(),
+        metadataResult.getErrorCount(),
+        calculateTimeTaken(conversionStartInstant));
+
     return metadataResult;
   }
 
-  private Pair<Map<Path, String>, Integer> getNewOrModifiedConfigFilesContents(
+  private MappedResults<SigningMetadata> convertConfigFileContent(
+      final Map<Path, String> contentMap, final SignerParser signerParser) {
+    final AtomicInteger errorCount = new AtomicInteger(0);
+    final List<SigningMetadata> signingMetadataList =
+        contentMap.entrySet().stream()
+            .flatMap(
+                entry -> {
+                  try {
+                    return signerParser.readSigningMetadata(entry.getValue()).stream();
```

Comment:
Does making the file loading sequential have an impact on the processing time for the key loading?

Reply:
currently the sequential loading is only performed for hashicorp conversions. The file reading is done in a separate step earlier (which is parallel). The conversion of loaded SigningMetadata -> SignerArtifact is sequential for Hashicorp, and parallel for all other types (including encrypted keystores).

Reply:
L109 is working on file content which is already loaded in memory. i.e. we are converting raw yaml text to SigningMetadadata object (using Jackson) ... all in-memory.

Reply:
The conversion from string to SigningMetadadata is now all done sequentially but before it was done in parallel as it was combined as part of the parse step. Has this had an impact on processing time? Most likely not the critical part of the loading though.

Looks like the file loading is done sequentially as far I can tell but that hasn't changed as part of this PR. So I guess we could improve that later.

Reply:
we should be able to convert `contentMap.entrySet().stream()` to `parallelStream()` without any issues.